### PR TITLE
Remove unused imports, and deprecated function usage

### PR DESCRIPTION
**Pull request by kdb424**
Compiler warned of unused imports.

### CODE REVIEW

File: compiler/docgen.nim [hunk 1]
```diff
@@ -835,15 +835,15 @@ proc docstringSummary(rstText: string): string =
   result = rstText.substr(2).strip
   var pos = result.find('\L')
   if pos > 0:
-    result.delete(pos, result.len - 1)
+    result = result[0 .. pos]
```
Comment (timotheecour) [inline compiler/docgen.nim]: buggy patch and allocates un-necessarily

Comment (kdb424) [reply compiler/docgen.nim]: My mistake on the allocation. Slices can be done in place, so my mistake on that. Can you elaborate on "buggy"? It should achieve the same result as far as I can tell.

Comment (timotheecour) [reply compiler/docgen.nim]: off by 1 error, try it

Comment (kdb424) [reply compiler/docgen.nim]: Thank you. I came from python, and read up on the differences. Thank you for your patience. 
EDIT: I'm reverting this change as even after getting the correct slice `result 0 ..< pos` I still could not avoid the assignment. Will simplify the PR to only deleting unused imports.

Comment (timotheecour) [reply compiler/docgen.nim]: why not simply use `result.setLen(pos - 1)` ?

Comment (kdb424) [reply compiler/docgen.nim]: That does in fact do that both in place, and do the same thing. Thank you again for helping me learn about these things, and being patient.

**Review by timotheecour — APPROVED** [review]

**Pull request merged**